File: implementations/elixir/ockam/ockam/lib/ockam/examples/echoer.ex
defmodule Ockam.Examples.Echoer do
  @moduledoc false

  use Ockam.Worker

  alias Ockam.Message
  alias Ockam.Worker

  @impl true
  def handle_message(message, state) do
    reply = Message.reply(message, state.address, Message.payload(message))

    Worker.route(reply, state)

    {:ok, state}
  end
end

File: implementations/rust/ockam/ockam_examples/example_projects/no_std/src/echoer.rs
#[cfg(all(not(feature = "std"), feature = "cortexm"))]
use ockam::{
    compat::{
        boxed::Box,
        string::String
    },
    println,
};
use ockam::{NodeContext, Result, Routed, Worker};

pub struct Echoer;

#[ockam::worker]
impl<C: NodeContext> Worker<C> for Echoer {
    type Message = String;

    async fn handle_message(&mut self, ctx: &mut C, msg: Routed<String>) -> Result<()> {
        println!("Address: {}, Received: {}", ctx.address(), msg);

        // Echo the message body back on its return_route.
        ctx.send(msg.return_route(), msg.body()).await
    }
}
